client/src/components/InlinePanel/index.js

```diff
@@ -315,4 +324,24 @@ export class InlinePanel extends ExpandingFormset {
       }),
     );
   }
+
+  /**
+   * Update fields based on the current DOM order.
+   */
+  updateOrderValues() {
```

Comment:
This could also be used for the "move up" and "move down" implementations, and save a fair bit of code.

Reply:
Sounds like a nice refactoring opportunity!

Reply:
Before refactoring we should consider the migration path towards Stimulus here. 

https://github.com/wagtail/wagtail/blob/main/client/src/controllers/FormsetController.ts has the initial implementation used in a few simpler inline panels.